### ISSUE TITLE
FEValues cleanup: Document arguments of FiniteElement::get_*_data()

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#1853]
#1277 addresses the issue of arguments of `FiniteElement::fill_fe_*_values()`. As I had previously already done for `Mapping::get_*_data()`, a similar cleanup should also happen for `FiniteElement::get_*_data()`.

In reference to #1198. Depends on #1851.
